TextFormat.md

```diff
@@ -43,43 +38,379 @@ support more human-readable representations, but never at the cost of accurate r
+
+```
+  (block $default
+    (block $green
+      (block $yellow
+        (block $orange
+          (block $red
+            (br_table (get_local $index), [$red, $orange, $yellow, $green] $default)
+          )
+          // ...
+        )
+        // ...
+      )
+      // ...
+    )
+    // ...
+  )
+```
+
+`br_table`s can have large numbers of labels, so this feature allows us to
+avoid very deep nesting in many cases.
+
+Note that when a label appears just before the closing `}`, it doesn't introduce
+a new block; it just provides a name for the enclosing block's label.
+
+## Operators with special syntax
+
+The default syntax for operators is their opcode name followed by a
+comma-separated list of operands. However, several operators have
+special syntax.
+
+
+### Control flow operators ([described here](https://github.com/WebAssembly/design/blob/master/AstSemantics.md))
+
+| Name | Syntax | Examples
+| ---- | ---- | ---- |
+| `block` | `{` … *label*: `}` | `{ br $a; a: }`
+| `loop` | `loop` *label* `{` … `}` | `loop $a { br $a; }`
+| `if` | `if` `(` *condition* `)` `{` … `}` | `if (0) { 1 }`
+| `if` | `if` `(` *condition* `) `{` … `} else `{` … `}` | `if (0) { 1 } else { 2 }`
+| `br` | `br` *label* | `br $where`
+| `br` | `br` `(` *expr* `)` *label* | `br ($v) $where`
+| `br_if` | `br_if` `(` *expr* `)` *label* | `br_if ($x < $y) $where`
+| `br_if` | `br_if` `(` *expr* `,` *condition* `)` *label* | `br_if ($v, $x < $y) $where`
+| `br_table` | `br_table` `(` *index-expr* `)` `[` *label* `,` … `]` `,` *default-label* | `br_table ($i) [$somewhere, $or_other], $default`
+| `br_table` | `br_table` `(` *expr* `,` *index-expr* `)` `[` *label* `,` … `]` `,` *default-label* | `br_table ($v, $i) [$somewhere, $or_other], $default`
+| `return` | `return` | `return`
+| `return` | `return` *expr* | `return $x`
+| `unreachable` | `unreachable` | `unreachable`
+
+
+### Basic operators ([described here](https://github.com/WebAssembly/design/blob/master/AstSemantics.md#constants))
+
+| Name | Syntax | Example
+| ---- | ---- | ---- |
+| `i32.const` | … | `234`, `0xfff7`
+| `i64.const` | … | `234`, `0xfff7`
+| `f64.const` | … | `0.1p2`, `infinity`, `nan:0x789`
+| `f32.const` | … | `0.1p2`, `infinity`, `nan:0x789`
+| `get_local` | *name* | `$x + 1`
+| `set_local` | *name* `=` *expr* | `$x = 1`
+| `select` | `select` *expr*, *expr*, *expr* | `select 1, 2, $x < $y`
+| `call` | `call` *name* `(`*expr* `,` … `)` | `call $min(0, 2)`
+| `call_import` | `call_import` *name* `(`*expr* `,` … `)` | `call_import $max(0, 2)`
+| `call_indirect` | `call_indirect` *signature-name* `[` *expr* `] (`*expr* `,` … `)` | `call_indirect $foo [1] (0, 2)`
+
+### Arithmetic operators ([described here](AstSemantics#32-bit-integer-operators))
+
+Arithmetic operators use C/JS-like infix and prefix notation.
+
+Add, sub, mul, div, rem, and, or, xor, shl, and shr operators use
+`+`, `-`, `*`, `/`, `%`, `&`, `|`, `^`, `<<`, and `>>`, respectively.
```

Comment:
Is there a reason why it would cause? (Assuming that majority of people familiar with C or JavaScript)

There is always an options to leave additional alternative for infix operators in form of function call similar to e.g. `i32.min`, so xor operator will have `i32.xor` in addition to `^`.